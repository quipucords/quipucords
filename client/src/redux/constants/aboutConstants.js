const ABOUT_DIALOG_OPEN = 'ABOUT_DIALOG_OPEN';
const ABOUT_DIALOG_CLOSE = 'ABOUT_DIALOG_CLOSE';

export { ABOUT_DIALOG_OPEN, ABOUT_DIALOG_CLOSE };
