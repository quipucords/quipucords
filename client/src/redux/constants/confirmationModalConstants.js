const CONFIRMATION_MODAL_SHOW = 'CONFIRMATION_MODAL_SHOW';
const CONFIRMATION_MODAL_HIDE = 'CONFIRMATION_MODAL_HIDE';

export { CONFIRMATION_MODAL_SHOW, CONFIRMATION_MODAL_HIDE };
