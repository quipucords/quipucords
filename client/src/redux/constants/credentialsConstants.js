
const LOAD_CREDENTIALS_SUCCESS = 'LOAD_CREDENTIALS_SUCCESS';

export {
  LOAD_CREDENTIALS_SUCCESS
}
