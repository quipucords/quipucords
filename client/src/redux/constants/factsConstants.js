const ADD_FACTS_ERROR = 'ADD_FACTS_ERROR';
const ADD_FACTS_LOADING = 'ADD_FACTS_LOADING';
const ADD_FACTS_SUCCESS = 'ADD_FACTS_SUCCESS';

export { ADD_FACTS_ERROR, ADD_FACTS_LOADING, ADD_FACTS_SUCCESS };
