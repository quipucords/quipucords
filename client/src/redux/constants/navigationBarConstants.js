const NAV_TOGGLE_COLLAPSE = 'NAV_TOGGLE_COLLAPSE';
const NAV_COLLAPSE = 'NAV_COLLAPSE';
const NAV_OPEN = 'NAV_OPEN';

export { NAV_TOGGLE_COLLAPSE, NAV_COLLAPSE, NAV_OPEN };
