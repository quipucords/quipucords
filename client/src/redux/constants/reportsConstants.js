const GET_REPORT = 'GET_REPORT';
const GET_MERGE_REPORT = 'GET_MERGE_REPORT';

export { GET_REPORT, GET_MERGE_REPORT };
