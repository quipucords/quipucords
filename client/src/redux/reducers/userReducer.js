import { userTypes } from '../constants';
import helpers from '../../common/helpers';

const initialState = {
  session: {
    error: false,
    errorMessage: '',
    pending: false,
    fulfilled: false,
    loggedIn: false,
    authToken: null,
    wasLoggedIn: false
  },
  user: {
    error: false,
    errorMessage: '',
    pending: false,
    fulfilled: false,
    currentUser: {}
  }
};

const userReducer = (state = initialState, action) => {
  switch (action.type) {
    case helpers.REJECTED_ACTION(userTypes.USER_INFO):
      return helpers.setStateProp(
        'user',
        {
          error: action.error,
          errorMessage: helpers.getMessageFromResults(action.payload)
        },
        {
          state,
          initialState
        }
      );

    case helpers.PENDING_ACTION(userTypes.USER_INFO):
      return helpers.setStateProp(
        'user',
        {
          pending: true
        },
        {
          state,
          initialState
        }
      );

    case helpers.FULFILLED_ACTION(userTypes.USER_INFO):
      return helpers.setStateProp(
        'user',
        {
          currentUser: action.payload.data,
          fulfilled: true
        },
        {
          state,
          initialState
        }
      );

    case helpers.REJECTED_ACTION(userTypes.USER_AUTH):
      return helpers.setStateProp(
        'session',
        {
          error: action.error,
          errorMessage: helpers.getMessageFromResults(action.payload),
          wasLoggedIn: state.session.wasLoggedIn
        },
        {
          state,
          initialState
        }
      );

    case helpers.PENDING_ACTION(userTypes.USER_AUTH):
      return helpers.setStateProp(
        'session',
        {
          pending: true,
          wasLoggedIn: state.session.wasLoggedIn
        },
        {
          state,
          initialState
        }
      );

    case helpers.FULFILLED_ACTION(userTypes.USER_AUTH):
      return helpers.setStateProp(
        'session',
        {
          loggedIn: true,
          fulfilled: true,
          wasLoggedIn: true,
          authToken: action.payload.authToken
        },
        {
          state,
          initialState
        }
      );

    case helpers.REJECTED_ACTION(userTypes.USER_LOGOUT):
      return helpers.setStateProp(
        'session',
        {
          error: action.error,
          errorMessage: helpers.getMessageFromResults(action.payload),
          wasLoggedIn: state.session.wasLoggedIn
        },
        {
          state,
          initialState
        }
      );

    case helpers.PENDING_ACTION(userTypes.USER_LOGOUT):
      return helpers.setStateProp(
        'session',
        {
          pending: true,
          wasLoggedIn: state.session.wasLoggedIn
        },
        {
          state,
          initialState
        }
      );

    case helpers.FULFILLED_ACTION(userTypes.USER_LOGOUT):
      return helpers.setStateProp(
        'session',
        {
          loggedIn: false,
          fulfilled: true,
          wasLoggedIn: state.session.wasLoggedIn
        },
        {
          state,
          initialState
        }
      );

    default:
      return state;
  }
};

userReducer.initialState = initialState;

export { userReducer as default, initialState, userReducer };
