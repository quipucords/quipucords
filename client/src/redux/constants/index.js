import * as aboutTypes from './aboutConstants';
import * as credentialsTypes from './credentialsConstants';
import * as factsTypes from './factsConstants';
import * as navigationBarTypes from './navigationBarConstants';
import * as reportsTypes from './reportsConstants';
import * as scansTypes from './scansConstants';
import * as sourcesTypes from './sourcesConstants';
import * as viewToolbarTypes from './viewToolbarConstants';

const reduxTypes = {
  about: aboutTypes,
  credentials: credentialsTypes,
  facts: factsTypes,
  navigation: navigationBarTypes,
  reports: reportsTypes,
  scans: scansTypes,
  sources: sourcesTypes,
  viewToolbar: viewToolbarTypes
};

export {
  reduxTypes,
  aboutTypes,
  credentialsTypes,
  factsTypes,
  navigationBarTypes,
  reportsTypes,
  scansTypes,
  sourcesTypes,
  viewToolbarTypes
};

export default reduxTypes;
