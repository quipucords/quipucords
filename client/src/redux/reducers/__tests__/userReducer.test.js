import helpers from '../../../common/helpers';
import { userTypes } from '../../constants/index';
import userReducer from '../userReducer';

const initialState = {
  session: {
    error: false,
    errorMessage: '',
    pending: false,
    fulfilled: false,
    loggedIn: false,
    authToken: null,
    wasLoggedIn: false
  },
  user: {
    error: false,
    errorMessage: '',
    pending: false,
    fulfilled: false,
    currentUser: {}
  }
};

describe('userReducer', () => {
  it('should return the initial state', () => {
    expect(userReducer(undefined, {})).toEqual(initialState);
  });

  it('should handle USER_INFO_REJECTED', () => {
    const dispatched = {
      type: helpers.REJECTED_ACTION(userTypes.USER_INFO),
      error: true,
      payload: {
        message: 'BACKUP MESSAGE',
        response: {
          data: {
            detail: 'USER INFO ERROR'
          }
        }
      }
    };

    const resultState = userReducer(undefined, dispatched);
    expect(resultState.user.error).toBeTruthy();
    expect(resultState.user.errorMessage).toEqual('USER INFO ERROR');

    expect(resultState.session).toEqual(initialState.session);
  });

  it('should handle USER_INFO_PENDING', () => {
    const dispatched = {
      type: helpers.PENDING_ACTION(userTypes.USER_INFO)
    };

    const resultState = userReducer(undefined, dispatched);

    expect(resultState.user.pending).toBeTruthy();
    expect(resultState.session).toEqual(initialState.session);
  });

  it('should handle USER_INFO_FULFILLED', () => {
    const dispatched = {
      type: helpers.FULFILLED_ACTION(userTypes.USER_INFO),
      payload: {
        data: {
          userName: 'admin',
          id: 1
        }
      }
    };

    const resultState = userReducer(undefined, dispatched);

    expect(resultState.user.fulfilled).toBeTruthy();
    expect(resultState.user.currentUser.userName).toEqual('admin');

    expect(resultState.session).toEqual(initialState.session);
  });

  it('should handle USER_AUTH_REJECTED', () => {
    const dispatched = {
      type: helpers.REJECTED_ACTION(userTypes.USER_AUTH),
      error: true,
      payload: {
        message: 'BACKUP MESSAGE',
        response: {
          data: {
            detail: 'USER AUTH ERROR'
          }
        }
      }
    };

    const resultState = userReducer(undefined, dispatched);
    expect(resultState.session.error).toBeTruthy();
    expect(resultState.session.errorMessage).toEqual('USER AUTH ERROR');

    expect(resultState.user).toEqual(initialState.user);
  });

  it('should handle USER_AUTH_PENDING', () => {
    const dispatched = {
      type: helpers.PENDING_ACTION(userTypes.USER_AUTH)
    };

    const resultState = userReducer(undefined, dispatched);

    expect(resultState.session.pending).toBeTruthy();
    expect(resultState.user).toEqual(initialState.user);
  });

  it('should handle USER_AUTH_FULFILLED', () => {
    const dispatched = {
      type: helpers.FULFILLED_ACTION(userTypes.USER_AUTH),
      payload: {
        authToken: 'spoof'
      }
    };

    const resultState = userReducer(undefined, dispatched);

    expect(resultState.session.fulfilled).toBeTruthy();
    expect(resultState.session.loggedIn).toBeTruthy();
    expect(resultState.session.wasLoggedIn).toBeTruthy();
    expect(resultState.session.authToken).toBeTruthy();
    expect(resultState.session.authToken).toEqual('spoof');

    expect(resultState.user).toEqual(initialState.user);
  });

  it('should handle USER_LOGOUT_REJECTED', () => {
    const dispatched = {
      type: helpers.REJECTED_ACTION(userTypes.USER_LOGOUT),
      error: true,
      payload: {
        message: 'BACKUP MESSAGE',
        response: {
          data: {
            detail: 'USER LOGOUT ERROR'
          }
        }
      }
    };

    const resultState = userReducer(undefined, dispatched);
    expect(resultState.session.error).toBeTruthy();
    expect(resultState.session.errorMessage).toEqual('USER LOGOUT ERROR');

    expect(resultState.user).toEqual(initialState.user);
  });

  it('should handle USER_LOGOUT_PENDING', () => {
    const dispatched = {
      type: helpers.PENDING_ACTION(userTypes.USER_LOGOUT)
    };

    const resultState = userReducer(undefined, dispatched);

    expect(resultState.session.pending).toBeTruthy();
    expect(resultState.user).toEqual(initialState.user);
  });

  it('should handle USER_LOGOUT_FULFILLED', () => {
    let dispatched = {
      type: helpers.FULFILLED_ACTION(userTypes.USER_AUTH),
      payload: {
        authToken: 'spoof'
      }
    };

    let resultState = userReducer(undefined, dispatched);

    expect(resultState.session.fulfilled).toBeTruthy();
    expect(resultState.session.loggedIn).toBeTruthy();
    expect(resultState.session.wasLoggedIn).toBeTruthy();
    expect(resultState.session.authToken).toBeTruthy();
    expect(resultState.session.authToken).toEqual('spoof');

    dispatched = {
      type: helpers.FULFILLED_ACTION(userTypes.USER_LOGOUT)
    };

    resultState = userReducer(resultState, dispatched);

    expect(resultState.session.fulfilled).toBeTruthy();
    expect(resultState.session.loggedIn).toBeFalsy();
    expect(resultState.session.wasLoggedIn).toBeTruthy();

    expect(resultState.user).toEqual(initialState.user);
  });
});
