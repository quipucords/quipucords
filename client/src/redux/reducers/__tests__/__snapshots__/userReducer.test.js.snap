// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fulfilled types USER_AUTH 1`] = `
Object {
  "result": Object {
    "session": Object {
      "authToken": undefined,
      "error": false,
      "errorMessage": "",
      "fulfilled": true,
      "loggedIn": true,
      "pending": false,
      "wasLoggedIn": true,
    },
    "user": Object {
      "currentUser": Object {},
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": false,
    },
  },
  "type": "USER_AUTH_FULFILLED",
}
`;

exports[`fulfilled types USER_INFO 1`] = `
Object {
  "result": Object {
    "session": Object {
      "authToken": null,
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "loggedIn": false,
      "pending": false,
      "wasLoggedIn": false,
    },
    "user": Object {
      "currentUser": Object {
        "test": "success",
      },
      "error": false,
      "errorMessage": "",
      "fulfilled": true,
      "pending": false,
    },
  },
  "type": "USER_INFO_FULFILLED",
}
`;

exports[`fulfilled types USER_LOGOUT 1`] = `
Object {
  "result": Object {
    "session": Object {
      "authToken": null,
      "error": false,
      "errorMessage": "",
      "fulfilled": true,
      "loggedIn": false,
      "pending": false,
      "wasLoggedIn": false,
    },
    "user": Object {
      "currentUser": Object {},
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": false,
    },
  },
  "type": "USER_LOGOUT_FULFILLED",
}
`;

exports[`pending types USER_AUTH 1`] = `
Object {
  "result": Object {
    "session": Object {
      "authToken": null,
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "loggedIn": false,
      "pending": true,
      "wasLoggedIn": false,
    },
    "user": Object {
      "currentUser": Object {},
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": false,
    },
  },
  "type": "USER_AUTH_PENDING",
}
`;

exports[`pending types USER_INFO 1`] = `
Object {
  "result": Object {
    "session": Object {
      "authToken": null,
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "loggedIn": false,
      "pending": false,
      "wasLoggedIn": false,
    },
    "user": Object {
      "currentUser": Object {},
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": true,
    },
  },
  "type": "USER_INFO_PENDING",
}
`;

exports[`pending types USER_LOGOUT 1`] = `
Object {
  "result": Object {
    "session": Object {
      "authToken": null,
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "loggedIn": false,
      "pending": true,
      "wasLoggedIn": false,
    },
    "user": Object {
      "currentUser": Object {},
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": false,
    },
  },
  "type": "USER_LOGOUT_PENDING",
}
`;

exports[`rejected types USER_AUTH 1`] = `
Object {
  "result": Object {
    "session": Object {
      "authToken": null,
      "error": true,
      "errorMessage": "ERROR",
      "fulfilled": false,
      "loggedIn": false,
      "pending": false,
      "wasLoggedIn": false,
    },
    "user": Object {
      "currentUser": Object {},
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": false,
    },
  },
  "type": "USER_AUTH_REJECTED",
}
`;

exports[`rejected types USER_INFO 1`] = `
Object {
  "result": Object {
    "session": Object {
      "authToken": null,
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "loggedIn": false,
      "pending": false,
      "wasLoggedIn": false,
    },
    "user": Object {
      "currentUser": Object {},
      "error": true,
      "errorMessage": "ERROR",
      "fulfilled": false,
      "pending": false,
    },
  },
  "type": "USER_INFO_REJECTED",
}
`;

exports[`rejected types USER_LOGOUT 1`] = `
Object {
  "result": Object {
    "session": Object {
      "authToken": null,
      "error": true,
      "errorMessage": "ERROR",
      "fulfilled": false,
      "loggedIn": false,
      "pending": false,
      "wasLoggedIn": false,
    },
    "user": Object {
      "currentUser": Object {},
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": false,
    },
  },
  "type": "USER_LOGOUT_REJECTED",
}
`;
